base/math.jl

```diff
@@ -91,7 +91,9 @@ deg2rad(z::Integer) = deg2rad(float(z))
 @vectorize_1arg Real rad2deg
 @vectorize_1arg Real deg2rad
 
-log(b,x) = log(x)./log(b)
+log{T<:Number}(b::T, x::T) = log(x)/log(b)
+log(b::Number, x::Number) = log(promote(b,x)...)
+@vectorize_2arg Number log
```

Comment:
Should we have tests for the vectorized version also, or is it enough that `@vectorize_2arg` is a well known macro that we expect to work.